mykaala/umassmarketplace
# Endpoints fix

This pull request attempts to add and serve a new product details page (productDetails.html) located in m3/product-details/. The expectation is that visiting http://localhost:3000/product-details/productDetails.html will serve that file directly. The PR also ensures that static files are served before hitting the wildcard route, allowing users to access specific pages without being redirected to home.html.

Changes Made:

Verified that productDetails.html is placed under m3/product-details/.
Ensured express.static middleware is applied before the wildcard catch-all route in App.js.
Confirmed that the correct URL path (http://localhost:3000/product-details/productDetails.html) is used to access the file.
Testing Steps:

Run the server with node App.js.
Navigate to http://localhost:3000/product-details/productDetails.html.
Expected Outcome:
The browser should display the product details page without redirecting to home.html.

Note:
This PR is likely to be rejected due to the current complexity of directory structure, lack of routing clarity, and potential issues with naming conventions or file placement. Additionally, passing full product JSON in query parameters may not meet best practices or team standards.
